### ISSUE TITLE
Some [Serial] tests are flaky

### ISSUE THREAD

**Issue by ihmccreery** [kubernetes/kubernetes#20230]
As part of #10548 I'm trying to run only `[Serial]` tests in serial and run the rest in parallel to speed up recovery time of the submit queue.  However, when doing so, the following arise:

- [ ] Etcd failure [Disruptive] should recover from SIGKILL (Failed 2 times in the last 30 runs. Stability: 93 %) No flake or failure
- [ ] Etcd failure [Disruptive] should recover from network partition with master (Failed 2 times in the last 30 runs. Stability: 93 %) No flake or failure
- [x] Kubelet [Serial] [Slow] regular resource usage tracking for 35 pods per node over 20m0s (Failed 2 times in the last 30 runs. Stability: 93 %)Failed with 0 retries (#20231)
- [ ] Nodes [Disruptive] Network when a node becomes unreachable [replication controller] recreates pods scheduled on the unreachable node AND allows scheduling of pods on a node after it rejoins the cluster (Failed 2 times in the last 30 runs. Stability: 93 %) No flake or failure
- [ ] Restart [Feature:Restart] should restart all nodes and ensure all nodes and pods recover (Failed 2 times in the last 30 runs. Stability: 93 %) No flake or failure
- [x] Kubelet [Serial] [Slow] regular resource usage tracking for 0 pods per node over 20m0s (Failed 3 times in the last 30 runs. Stability: 90 %)Failed with 0 retries (#20231)
- [ ] SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run [Conformance] (Failed 3 times in the last 30 runs. Stability: 90 %) No flake or failure
- [ ] Nodes [Disruptive] Resize [Slow] should be able to add nodes (Failed 4 times in the last 30 runs. Stability: 86 %) No flake or failure
- [ ] Nodes [Disruptive] Resize [Slow] should be able to delete nodes (Failed 5 times in the last 30 runs. Stability: 83 %) No flake or failure
- [ ] SchedulerPredicates [Serial] validates MaxPods limit number of pods that are allowed to run [Slow] (Failed 5 times in the last 30 runs. Stability: 83 %) No flake or failure

The job is http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-serial/

These tests seem way flakier in this job than when we run them in http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce/, but I have no idea why.  I'll continue working on triaging, but I thought I'd file an issue so @dchen1107 could start digging too.

**Comment by ihmccreery** [comment]
Ah, hm.  So, the "Kubelet [Serial] [Slow] ... resource usage tracking" tests aren't running in http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-slow/, and they're flaking/failing there too.  Filing a separate issue about that.

**Comment by ihmccreery** [comment]
Filed (#20231) for resource usage tracking issues.

**Comment by dchen1107** [comment]
I quickly scanned through all failures, and I highly suspected all failures are caused by heapster crashloop introduced by #20156 

kubernetes-e2e-gce-serial/312 is the current build, and pick up the revert pr https://github.com/kubernetes/kubernetes/pull/20215. We will see the result.

**Comment by dchen1107** [comment]
Ok kubernetes-e2e-gce-serial/312 is finished with one failure:

Reboot [Disruptive] each node by dropping all inbound packets for a while and ensure they function afterwards

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/reboot.go:116 Jan 27 12:31:24.793: Test failed; at least one node failed to reboot in the time given.

Still looking into why?

**Comment by dchen1107** [comment]
Ok, here is the reason of kubernetes-e2e-gce-serial/312 failure: influxdb container in 
monitoring-influxdb-grafana-v2-z7sbj is in crashloop: cc/ @yujuhong somehow I didn't see container logs being collected from failed pod here: 

```
12:31:25 Jan 27 12:31:25.150: INFO: event for monitoring-influxdb-grafana-v2-z7sbj: {kubelet e2e-serial-minion-tvkh} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "influxdb" with CrashLoopBackOff: "Back-off 2m40s restarting failed container=influxdb pod=monitoring-influxdb-grafana-v2-z7sbj_kube-system(6565af1a-c52b-11e5-a17c-42010af00002)"
...
12:31:25 Jan 27 12:31:25.165: INFO: monitoring-influxdb-grafana-v2-z7sbj          e2e-serial-minion-tvkh  Running         [{Ready False 0001-01-01 00:00:00 +0000 UTC 2016-01-27 12:26:24 -0800 PST ContainersNotReady containers with unready status: [influxdb]}]
```

Luckily I have my local cluster which also run into influxdb-crashloop issue, and I ssh-ed into the node and debugged why: 

1) Find previous dead influxdb container and manually start it and found the container exit immediately without any error message: 
```
# docker ps -a | grep influx
67d39cc50147        gcr.io/google_containers/heapster_influxdb:v0.4                        "influxdb --config /c"   3 minutes ago       Exited (1) 3 minutes ago                                                     k8s_influxdb.238041e4_monitoring-influxdb-grafana-v2-tggih_kube-system_5301b6df-c48b-11e5-985d-42010af00002_f64f889f
276f22c1964c        gcr.io/google_containers/heapster_influxdb:v0.4                        "influxdb --config /c"   8 minutes ago       Exited (1) 8 minutes ago                                                     k8s_influxdb.238041e4_monitoring-influxdb-grafana-v2-tggih_kube-system_5301b6df-c48b-11e5-985d-42010af00002_9dc83c4e
...
# docker start 67d39cc50147 
# docker inspect 67d39cc50147 
[
{
    "Id": "67d39cc501472d81cc3d7002dd91c9b6df215cbc88999cdbb300bb599945bf24",
    "Created": "2016-01-27T22:01:17.00738841Z",
    "Path": "influxdb",
    "Args": [
        "--config",
        "/config/config.toml"
    ],
    "State": {
        "Running": false,
        "Paused": false,
        "Restarting": false,
        "OOMKilled": false,
        "Dead": false,
        "Pid": 0,
        "ExitCode": 1,
        "Error": "",
        "StartedAt": "2016-01-27T22:04:55.978728847Z",
        "FinishedAt": "2016-01-27T22:04:56.030173992Z"
    },
...

# docker logs 67d39cc50147 

+---------------------------------------------+
|  _____        __ _            _____  ____   |
| |_   _|      / _| |          |  __ \|  _ \  |
|   | |  _ __ | |_| |_   ___  _| |  | | |_) | |
|   | | | '_ \|  _| | | | \ \/ / |  | |  _ <  |
|  _| |_| | | | | | | |_| |>  <| |__| | |_) | |
| |_____|_| |_|_| |_|\__,_/_/\_\_____/|____/  |
+---------------------------------------------+

[01/27/16 22:01:17] [INFO] Loading configuration file /config/config.toml
```

2) Tried to start a new container with same docker image and same mounts. The container exit immediately again without any error logging:

```
 docker run -p 8081:8083 -p 8082:8086 -v /var/lib/kubelet/pods/5301b6df-c48b-11e5-985d-42010af00002/volumes/kubernetes.io~empty-dir/influxdb-persistent-storage:/data gcr.io/google_containers/heapster_influxdb:v0.4 
```

3) Vishnu suggested it might be caused by the corrupted checkpoint of influxdb. So we removed all files under /data. Now influxdb coming up without any issue.

This is not the first time I saw influxdb crashloop which causes test failure. Should we remove influxdb-grafana-controller completely from our kube-system for e2e test here? I will file a separate issue for this.

**Comment by ihmccreery** [comment]
> This is not the first time I saw influxdb crashloop which causes test failure. Should we remove influxdb-grafana-controller completely from our kube-system for e2e test here? I will file a separate issue for this.

Per usual, I'll make the case that our e2e environment should look as similar to prod as possible; if that pod is running in prod systems, it should be running here.  (If it's not, why is it running in e2e?)

**Comment by yujuhong** [comment]
> @yujuhong somehow I didn't see container logs being collected from failed pod here:

The PR only print the logs if the pod becomes not ready *after* the test. I'll send another PR to print the logs for not ready pods even before the test start shortly.

**Comment by dchen1107** [comment]
@ihmccreery Honestly I want to stop having monitoring-influxdb-grafana in prod by default anyway, but of course I cannot make decision by myself here. cc/ @brendandburns @vishh @bgrant0607 @thockin 

We did poor job on supporting this, and we don't have much control on it too. Most of our users use different backend storage and monitoring tool. This one now only adds burden to us.

**Comment by yujuhong** [comment]
> The PR only print the logs if the pod becomes not ready after the test. I'll send another PR to print the logs for not ready pods even before the test start shortly.

FYI, #20248

**Comment by vishh** [comment]
+1 for not running InfluxDB + Grafana by default moving forward. We have a
few monitoring vendors that have integrated with k8s very well. We can move
the InfluxDB+Grafana support to contrib maybe?
Heapster is still required for the cluster and it can be tested as a
standalone entity.

**Comment by bgrant0607** [comment]
I like the idea of moving InfluxDB+Grafana to contrib.

**Comment by ihmccreery** [comment]
@vishh Please file a bug on removing InfluxDB + Grafana?  These tests seem to have healed with the Heapster fix, so I'm going to close this.

**Comment by dchen1107** [comment]
I filed  #20247 aleady.
